Aligning exception delivered to callback behaviour of client with server

sancar opened the pull request:
`ExecutionCallback.onFailure` registered to `ICompletableFuture.andThen`
 was wrapped to ExecutionException on client while on server exception
 is directly passed to `onFailure` without wrapping.

 Client behaviour is changed to behave same servers.

fixes #10335

devOpsHazelcast commented:
Test `FAIL`ed.
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8218/

devOpsHazelcast commented:
Test `FAIL`ed.
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8222/

devOpsHazelcast commented:
Test PASSed.

sancar commented:
@pveentjer 
After reading your comment I realised issue title could be misleading.  `resolve` method is not really same with server. Client `resolve` method is actually no-op after this change on client.  
I have updated description and added test to make the intention clearer.

This part come to my attention because of another issue I am working on https://github.com/hazelcast/hazelcast/issues/10291

pveentjer commented on hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationFuture.java:
```diff
@@ -63,19 +63,6 @@ protected TimeoutException newTimeoutException(long timeout, TimeUnit unit) {
-    }
-
-    @Override
-    protected Object resolve(Object value) {
```
This is the good part from the client: wrapping the remote exception in a generic exception. This is how it should be done in member/member in hz4.

sancar commented:
run-lab-run

devOpsHazelcast commented:
Test `FAIL`ed.
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8264/

sancar commented:
run-lab-run

sancar commented:
run-lab-run

sancar commented:
run-lab-run

sancar commented:
run-lab-run

devOpsHazelcast commented:
Test `FAIL`ed.
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8277/

devOpsHazelcast commented:
Test `FAIL`ed.
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8278/

sancar commented:
run-lab-run

devOpsHazelcast commented:
Test PASSed.
